$OpenBSD: patch-make_hotspot_lib_JvmOverrideFiles_gmk,v 1.1 2019/07/23 11:55:42 kurt Exp $

Reduce optimization level for parse2.cpp on x86 with clang:
* jtreg test hotspot/jtreg/compiler/c2/Test8062950.java fails when
  parse2.cpp is optimized above -O1 with clang on x86.

Index: make/hotspot/lib/JvmOverrideFiles.gmk
--- make/hotspot/lib/JvmOverrideFiles.gmk.orig
+++ make/hotspot/lib/JvmOverrideFiles.gmk
@@ -210,6 +210,14 @@ else ifeq ($(OPENJDK_TARGET_OS), bsd)
   endif
 
   ifeq ($(TOOLCHAIN_TYPE), clang)
+    ifeq ($(OPENJDK_TARGET_CPU), x86)
+      ifneq ($(DEBUG_LEVEL), slowdebug)
+        # hotspot/jtreg/compiler/c2/Test8062950.java test fails on x86
+        # with clang when parse2.cpp is optimized above -O1
+        BUILD_LIBJVM_parse2.cpp_CXXFLAGS := -O1
+      endif
+    endif
+
     # The following files are compiled at various optimization
     # levels due to optimization issues encountered at the
     # default level. The Clang compiler issues a compile
@@ -224,7 +232,8 @@ else ifeq ($(OPENJDK_TARGET_OS), bsd)
         sharedRuntimeTrig.cpp \
         sharedRuntimeTrans.cpp \
         loopTransform.cpp \
-        unsafe.cpp
+        unsafe.cpp \
+        parse2.cpp \
         #
   endif
 
